spec/models/course/condition/assessment_spec.rb

```diff
@@ -7,6 +7,43 @@
   with_tenant(:instance) do
     let(:course) { create(:course) }
 
+    describe 'validations' do
+      context 'when an assessment is its own condition' do
+        subject do
+          assessment = create(:assessment, course: course)
+          build_stubbed(:assessment_condition,
+                        course: course, assessment: assessment, conditional: assessment).
+            tap do |assessment_condition|
+            allow(assessment_condition).to receive(:assessment_id_changed?).and_return(true)
+          end
+        end
+        it { is_expected.to_not be_valid }
+      end
+
+      context "when an assessment is already included in its conditional's conditions" do
+        subject do
+          create(:assessment_condition, course: course).tap do |assessment_condition|
+            allow(assessment_condition).to receive(:assessment_id_changed?).and_return(true)
```

Comment:
Change the assessment_id of this assessment condition would require to add another assessment condition to the same conditional.